androidHyperskillApp/src/main/java/org/hyperskill/app/android/home/view/ui/fragment/HomeFragment.kt

```diff
@@ -97,17 +101,23 @@ class HomeFragment :
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        setProblemsLimitFragment()
         initViewStateDelegate()
         initGamificationToolbarDelegate()
         problemOfDayCardFormDelegate.setup(viewBinding.homeScreenProblemOfDayCard)
         topicsToDiscoverNextDelegate.setup(
             requireContext(),
             viewBinding.homeTopicsToDiscoverNext.homeTopicsToDiscoverNextRecycler
         )
+        problemsLimitDelegate = ProblemsLimitDelegate(
+            viewBinding = viewBinding.homeProblemsLimit,
+            onNewMessage = {
+                homeViewModel.onNewMessage(HomeFeature.Message.ProblemsLimitMessage(it))
+            }
+        )
+        problemsLimitDelegate?.setup(context = requireContext())
         with(viewBinding) {
-            homeScreenSwipeRefreshLayout.setHyperskillColors()
-            homeScreenSwipeRefreshLayout.setOnRefreshListener {
+            root.setHyperskillColors()
+            root.setOnRefreshListener {
                 homeViewModel.onNewMessage(HomeFeature.Message.PullToRefresh)
             }
             homeScreenError.tryAgain.setOnClickListener {
```

Comment:
ProblemsLimitDelegate hold a reference to viewBinding, that's why it should be destroyed, so that view reference will not be hold after onDestroyView hook